src/term.c

```diff
@@ -287,17 +287,23 @@ static int cmd_dump(PROGRAMMER *pgm, AVRPART *p, int argc, char *argv[]) {
   // Get start address if present
   char *end_ptr;
   if (argc >= 3 && strcmp(argv[2], "...") != 0) {
-    unsigned long ul = strtoul(argv[2], &end_ptr, 0);
+    int addr = strtol(argv[2], &end_ptr, 0);
     if(*end_ptr || (end_ptr == argv[2])) {
       pmsg_error("(dump) cannot parse address %s\n", argv[2]);
       return -1;
     }
-    if(ul > INT_MAX || ul >= (unsigned long) maxsize) {
-      pmsg_error("(dump) %s address 0x%lx is out of range [0, 0x%0*x]\n", mem->desc, ul,
-        mem->size > 0x10000? 5: 4, maxsize-1);
+
+    // Turn negative addr value (counting from top and down) into an actual memory address
+    if (addr < 0)
+      addr = maxsize + addr;
+
+    if (addr < 0 || addr >= maxsize) {
+      int digits = mem->size > 0x10000? 5: 4;
+      pmsg_error("(dump) %s address %s is out of range [-0x%0*x, 0x%0*x]\n",
+        mem->desc, argv[2], digits, maxsize, digits, maxsize-1);
```

Comment:
I like that the valid address range now reads [-maxsize, maxsize-1]; I hadn't thought of that, good.